cmd/addon-operator-manager/main.go

```diff
@@ -71,10 +73,12 @@ func initReconcilers(mgr ctrl.Manager) {
 		os.Exit(1)
 	}
 
+	recorder := metrics.NewRecorder(true)
```

Comment:
Would you mind adding a flag `enable-metrics-recorder` to switch this on/off?
(Can be `true` by default)

We don't always need metrics when developing or we might have to run the Addon Operator in environments that don't have monitoring setup :)